gas-bound-caller/README.md

```diff
@@ -0,0 +1,47 @@
+# GasBoundCaller
+
+Starting from v24 On Era, the gas for pubdata is charged at the end of the execution of the entire transaction. This means that if a subcall is not trusted, it can consume a significant amount of pubdata during the process. While this may not be an issue for most contracts, there are use cases, e.g., for relayers, where it is crucial to ensure that the subcall will not spend more money than intended.
+
+The `GasBoundCaller` is a contract with the following interface:
+
+```solidity
+function gasBoundCall(address _to, uint256 _maxTotalGas, bytes calldata _data) external payable
+```
+
+> Note that the amount of gas passed into this function should be less than or equal to `_maxTotalGas`. If the computational gas provided is higher than `_maxTotalGas`, the higher value will be used.
+
+This contract will call the address `_to` with the entire execution gas passed to it, while ensuring that the total consumed gas does not exceed `_maxTotalGas` under any circumstances.
+
+If the call to the `_to` address fails, the gas used on pubdata is considered zero, and the total gas used is fully equivalent to the gas consumed within the execution. The `GasBoundCaller` will relay the revert message as-is.
+
+If the call to the `_to` address succeeds, the `GasBoundCaller` will ensure that the total consumed gas does not exceed `_maxTotalGas`. If it does, it will revert with a "Not enough gas for pubdata" error. If the total consumed gas is less than or equal to `_maxTotalGas`, the `GasBoundCaller` will return returndata equal to `abi.encode(bytes returndataFromSubcall, uint256 gasUsedForPubdata)`.
+
+## Usage
```

Comment:
Another legit and IMO most needed usage for this contract would be inheriting it and using it directly from the user's contract. This is needed to preserve context variables (such as `msg.sender`). For this, you need to make `gasBoundCall` visibility to be `public`.

Reply:
Discussed internally, it should be kept external and used as `this.gasBoundCall` instead